### PR TITLE
Update login functionality, delete servlets, display servlet and file…

### PR DESCRIPTION
**Pull request by Chronos34**
… permission

### CODE REVIEW

**Review by CharlesJ10 — APPROVED** [review]

File: capstone/src/main/java/com/google/sps/servlets/APKUploadServlet.java [hunk 1]
```diff
@@ -61,6 +73,10 @@
   public void doPost(HttpServletRequest request, HttpServletResponse response)
    throws ServletException, IOException {
 
+    if (!userService.isUserLoggedIn()) {return;}
```
Comment (HallM) [inline capstone/src/main/java/com/google/sps/servlets/APKUploadServlet.java]: You'll want to send a response back of sorts. This I believe would return an error that to the user would say something like request closed. Sending back a code 403 would be a better user experience to let the user know the issue is they are not logged in.

File: capstone/src/main/java/com/google/sps/servlets/FileDeletionServlet.java [hunk 2]
```diff
@@ -46,36 +51,93 @@
...
    throws IOException {
 
+    // The two lines below retrieve certain details about the desired apk
+    // so the wrong files aren't deleted.
     fileName = request.getParameter("file_name");
+    fileVisibility = request.getParameter("ownership");
+
+    // The condition below is a safety belt for situations involving
+    // the manipulation of permissions for public APKs
+    if (!userService.isUserLoggedIn()) {return;}
```
Comment (HallM) [inline capstone/src/main/java/com/google/sps/servlets/FileDeletionServlet.java]: Same 403 here.

File: capstone/src/main/java/com/google/sps/servlets/FileDeletionServlet.java [hunk 3]
```diff
@@ -46,36 +51,93 @@
...
+
+    currentUser = userService.getCurrentUser();
+
+    // This condition below is a second safety belt that ensures the requested
+    // file targeted for deletion belongs to the user making the request.
+    if (!deleteUnzippedApk(dataStore, fileName, currentUser.getUserId())) {return;}
```
Comment (HallM) [inline capstone/src/main/java/com/google/sps/servlets/FileDeletionServlet.java]: Similar, but I would use a 404 here.

File: capstone/src/main/java/com/google/sps/servlets/FileRetrievalServlet.java [hunk 4]
```diff
@@ -20,57 +20,130 @@
...
     Gson gson = new Gson();
 
     response.setContentType("application/json;");
     response.getWriter().println(gson.toJson(apks));
 
   }
+
+  private PreparedQuery retrievePrivateFiles(UserService userServices, DatastoreService datastorage) {
+
+    if (!userServices.isUserLoggedIn()) {return datastorage.prepare(new Query("Empty"));}
```
Comment (HallM) [inline capstone/src/main/java/com/google/sps/servlets/FileRetrievalServlet.java]: Another option could be to use Java's Optional class. The return could be an Optional<PreparedQuery>, then you'd return either
`Optional.empty()`
or
`Optional.of(datastorage.prepare(query));`
depending on if the user is logged in or not.

Then the use of this function above can skip trying to process entities.

File: capstone/src/main/java/com/google/sps/servlets/SignInServlet.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+  private UserService userService = UserServiceFactory.getUserService();
+
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+
+    if (!userService.isUserLoggedIn()) {
+      response.sendRedirect(userService.createLoginURL("/signin"));
+    } 
+    
+    if (userService.isUserLoggedIn()) {response.sendRedirect("/#/explore");}
```
Comment (HallM) [inline capstone/src/main/java/com/google/sps/servlets/SignInServlet.java]: Can use an else block or a similar style to the doPost where it does the condition and returns. The remainder of the function then can assume the logged in state.

**Review by HallM — CHANGES_REQUESTED** [review]

**Review by HallM — APPROVED** [review]

**Pull request merged**